### ISSUE TITLE
[Feature request] limit per-file ignores to a specific count

### ISSUE THREAD

**Issue by greg0ire** [vimeo/psalm#9127]
In https://github.com/doctrine/orm/pull/10415, while removing an ignore rule that was added for one occurrence of `ArgumentTypeCoercion`, I noticed we now had 5 of those. Because of this and of https://github.com/vimeo/psalm/issues/9102, I'm going to discuss internally about switching to `@psalm-suppress`, but it would be nice to have a syntax forbidding more than 1 occurence of a given type of error. It would be logical to use the same `occurrences` attribute as in the baseline file.

An alternative would be to allow having several baseline files, only one of which would be written to by `--set-baseline` (which needs an argument anyway).

**Comment by orklah** [comment]
you can completely have multiple baselines. As you say, you must specify which one you override and you can choose which one you use through command line (--use-baseline)

**Comment by greg0ire** [comment]
:thinking: but then you are only using one baseline at a give time, right?

**Comment by mpdude** [comment]
First, question: Is the baseline file intended to be updated repeatedly to suppress errors? My understanding was that it should be created once initially (to set everything to zero, thus "baseline") and then deal with necessary exceptions that come up later either with `@psalm-suppress` and/or `psalm.xml`.

Then, regarding the feature request: When an allowed number of occurences is given, it should be an error when the actual number _decreases_. That means you've fixed something and the config should be updated.

Still, there is a loophole left when you fix the problem in one place and add the same issue in another place on the same file, right?

**Comment by greg0ire** [comment]
> Still, there is a loophole left when you fix the problem in one place and add the same issue in another place on the same file, right?

Yes, and I think baseline file address this properly by mentioning the offending variable

**Comment by ygottschalk** [comment]
We could use xinclude to have multiple baseline files. This can be achieved by adding these two lines of code
```php
$baselineDoc->documentURI = $baselineFile;
$baselineDoc->xinclude();
```
here: https://github.com/vimeo/psalm/blob/aec0edc618674d001f5f653494f188269a7a6aa5/src/Psalm/ErrorBaseline.php#L90
